src/SFA.DAS.EAS.Web.IntegrationTests/TestUtils/ApiTester/ApiIntegrationTester.cs

```diff
@@ -0,0 +1,263 @@
+        {
+            var callResponse = new CallResponse();
+            await FetchInitialResponseAsync(call, callResponse);
+            return callResponse;
+        }
+
+        private async Task<CallResponse<TResult>> GetResponseAsync<TResult>(CallRequirements call)
+        {
+            var callResponse = new CallResponse<TResult>();
+            await FetchInitialResponseAsync(call, callResponse);
+            await FetchCompleteResponseAsync(callResponse);
+            return callResponse;
+        }
+
+        private async Task FetchInitialResponseAsync(CallRequirements call, CallResponse response)
+        {
+            EnsureStarted();
+            ClearDownForNewTest();
+            await MakeCall(call, response);
+            CheckCallWasSuccessful(call, response);
+        }
+
+        private async Task MakeCall(CallRequirements call, CallResponse response)
+        {
+            var client = GetClient(call);
+            response.Response = await client.GetAsync(call.Uri);
+        }
+
+        private async Task FetchCompleteResponseAsync<TResult>(CallResponse<TResult> response)
+        {
+            var responseContent = await response.Response.Content.ReadAsStringAsync();
+            response.Data = JsonConvert.DeserializeObject<TResult>(responseContent);
+        }
+
+        private HttpClient GetClient(CallRequirements call)
+        {
+            var client = TestServer.HttpClient;
+            client.Timeout = call.TimeOut;
+            return client;
+        }
+
+        private void ClearDownForNewTest()
+        {
+            _exceptionHandler.ClearException();
+            _dependencyResolver.ClearCreationLog();
+        }
+
+        private void EnsureStarted()
+        {
+            if (!IsTestServerStarted)
+            {
+                BeginTests();
+            }
+        }
+
+        private void InitialiseHost(IAppBuilder app)
+        {
+            HttpConfiguration config = new HttpConfiguration();
+            WebApiConfig.Register(config);
+            CustomiseIoC(config);
+            app.UseWebApi(config);
+        }
+
+        private void CustomiseIoC(HttpConfiguration configuration)
+        {
+            var existingExceptionHandler = configuration.Services.GetExceptionHandler();
+            _exceptionHandler = new IntegrationTestExceptionHandler(existingExceptionHandler);
+            configuration.Services.Replace(typeof(IAssembliesResolver), new TestWebApiResolver<AccountLegalEntitiesController>());
+            configuration.Services.Replace(typeof(IExceptionHandler), _exceptionHandler);
+
+            StructuremapMvc.Start();
+            StructuremapWebApi.Start();
+
+            var container = ((StructureMapWebApiDependencyResolver)GlobalConfiguration.Configuration.DependencyResolver).Container;
+
+            var requestContextMock = new Mock<IRequestContext>();
+
+            container.Configure(c =>
```

Comment:
I'd recommend a StructureMap registry specifically for the `SFA.DAS.EAS.Account.API.IntegrationTests` project. This will allow them to be defined in one place rather than across `ApiIntegrationTester` & `DbBuilderDependentRepositories` etc. You'll also be able to specify their lifetimes which should remove the need to define them as `Lazy`. Unless I'm missing something?